LibreNMS/OS/Shared/Cisco.php

```diff
@@ -396,15 +396,15 @@ public function pollNac()
                     'port_id' => $ifIndex_map->get($ifIndex, 0),
                     'mac_address' => $mac_address,
                     'auth_id' => $auth_id,
-                    'domain' => $portAuthSessionEntryParameters['cafSessionDomain'],
-                    'username' => $portAuthSessionEntryParameters['cafSessionAuthUserName'],
-                    'ip_address' => (string) IP::fromHexString($portAuthSessionEntryParameters['cafSessionClientAddress'], true),
-                    'host_mode' => $portAuthSessionEntryParameters['cafSessionAuthHostMode'],
-                    'authz_status' => $portAuthSessionEntryParameters['cafSessionStatus'],
-                    'authz_by' => $portAuthSessionEntryParameters['cafSessionAuthorizedBy'],
-                    'timeout' => $portAuthSessionEntryParameters['cafSessionTimeout'],
-                    'time_left' => $portAuthSessionEntryParameters['cafSessionTimeLeft'],
-                    'vlan' => $portAuthSessionEntryParameters['cafSessionAuthVlan'],
+                    'domain' => $portAuthSessionEntryParameters['cafSessionDomain'] ?? '',
```

Comment:
These should probably be this:
```suggestion
                    'domain' => $portAuthSessionEntryParameters['cafSessionDomain'] ?? null,
```

Reply:
(except the IP one)

Reply:
Not so sure about null here. 
```
MariaDB [librenms]> describe ports_nac;
+--------------+------------------+------+-----+---------+----------------+
| Field        | Type             | Null | Key | Default | Extra          |
+--------------+------------------+------+-----+---------+----------------+
| ports_nac_id | int(10) unsigned | NO   | PRI | NULL    | auto_increment |
| auth_id      | varchar(50)      | NO   |     | NULL    |                |
| device_id    | int(10) unsigned | NO   | MUL | NULL    |                |
| port_id      | int(10) unsigned | NO   | MUL | NULL    |                |
| domain       | varchar(50)      | NO   |     | NULL    |                |
| username     | varchar(50)      | NO   |     | NULL    |                |
| mac_address  | varchar(50)      | NO   |     | NULL    |                |
| ip_address   | varchar(50)      | NO   |     | NULL    |                |
| host_mode    | varchar(50)      | NO   |     | NULL    |                |
| authz_status | varchar(50)      | NO   |     | NULL    |                |
| authz_by     | varchar(50)      | NO   |     | NULL    |                |
| authc_status | varchar(50)      | NO   |     | NULL    |                |
| method       | varchar(50)      | NO   |     | NULL    |                |
| timeout      | varchar(50)      | NO   |     | NULL    |                |
| time_left    | varchar(50)      | YES  |     | NULL    |                |
| vlan         | int(10) unsigned | YES  |     | NULL    |                |
| time_elapsed | varchar(50)      | YES  |     | NULL    |                |
| created_at   | timestamp        | YES  |     | NULL    |                |
| updated_at   | timestamp        | YES  |     | NULL    |                |
| historical   | tinyint(1)       | NO   |     | 0       |                |
+--------------+------------------+------+-----+---------+----------------+
```
Only a few of them are nullable. I'll change them

Reply:
bizarre with the default value there, sounds good.